### PR TITLE
[ElementwiseOpToLLVM] Propagate Inf/NaN in fp8e4m3 / fp8e5m2 -> bf16 conversion

### LINKED ISSUE #7771: fp8e4m3 / fp8e5m2 -> bf16 conversion silently decodes NaN/Inf as finite values
### Summary

The FP8 → BF16 converters in `third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp` decode the reserved, non-finite fp8 encodings as finite bf16 values instead of propagating NaN/Inf. A NaN or Inf input is silently turned into a large finite number, with no diagnostic.

- **FP8E4M3FN → BF16:** the two reserved NaN encodings (bytes `0x7F`, `0xFF`) decode to ±480.0.
- **FP8E5M2 → BF16:** the Inf encodings (`0x7C`, `0xFC`) and the NaN encodings (`0x7D`-`0x7F`, `0xFD`-`0xFF`) all decode to finite values.

Both output paths are affected for each format:

- `Fp8E4M3Nv_to_Bf16` / `Fp8E5M2_to_Bf16` — the `fmul` rebias path, used on all non-LTS drivers
- `Fp8E4M3Nv_to_Bf16Table` / `Fp8E5M2_to_Bf16Table` — the LTS lookup-table path

The `→ FP16` converters in the same file do **not** have this problem, so the fate of a non-finite fp8 input currently depends on which output dtype you cast to. Any kernel relying on NaN/Inf propagation through an fp8 → bf16 cast gets a wrong answer. NaN-based masking is the common case: an element intended to be NaN becomes a finite value and participates in the computation. Because the substituted values are finite and moderate in magnitude, the result looks plausible rather than obviously broken.

### Reproducer

```python
import torch, triton, triton.language as tl

@triton.jit
def cast(src, dst, N: tl.constexpr):
    i = tl.arange(0, N)
    tl.store(dst + i, tl.load(src + i).to(dst.dtype.element_ty))

def run(fp8_dtype, bytes_):
    src = torch.tensor(bytes_, dtype=torch.uint8, device="xpu")
    dst = torch.empty(len(bytes_), dtype=torch.bfloat16, device="xpu")
    cast[(1,)](triton.reinterpret(src, fp8_dtype), dst, N=len(bytes_))
    torch.xpu.synchronize()
    bits = dst.view(torch.uint16).tolist()
    for i, byte in enumerate(bytes_):
        v = dst[i]
        print(f"byte 0x{byte:02X} -> 0x{bits[i]:04x}  value={float(v.item()):<12} "
              f"isnan={v.isnan().item()}  isinf={v.isinf().item()}")

print("fp8e4m3 -> bf16:")
run(torch.float8_e4m3fn, [0x7F, 0xFF])
print("fp8e5m2 -> bf16:")
run(torch.float8_e5m2, [0x7C, 0x7D, 0x7E, 0x7F, 0xFC, 0xFF])
```

Actual output on Intel Arc B580 (BMG):

```
fp8e4m3 -> bf16:
byte 0x7F -> 0x43f0  value=480.0        isnan=False  isinf=False   (should be NaN)
byte 0xFF -> 0xc3f0  value=-480.0       isnan=False  isinf=False   (should be NaN)
fp8e5m2 -> bf16:
byte 0x7C -> 0x4780  value=65536.0      isnan=False  isinf=False   (should be +Inf)
byte 0x7D -> 0x47a0  value=81920.0      isnan=False  isinf=False   (should be NaN)
byte 0x7E -> 0x47c0  value=98304.0      isnan=False  isinf=False   (should be NaN)
byte 0x7F -> 0x47e0  value=114688.0     isnan=False  isinf=False   (should be NaN)
byte 0xFC -> 0xc780  value=-65536.0     isnan=False  isinf=False   (should be -Inf)
byte 0xFF -> 0xc7e0  value=-114688.0    isnan=False  isinf=False   (should be NaN)
```

Note that the existing golden tests (`python/test/unit/intel/test_fp8e4m3_to_bf16.py`, `test_fp8e5m2_to_bf16.py`) assert these finite values and their docstrings recorded the behavior as intentional, so this was pinned rather than accidentally regressed.

### PR DESCRIPTION
**Pull request by etiotto**
The reserved fp8 encodings were decoded to finite bf16 values instead of propagating NaN/Inf. Both bf16 output paths (the `fmul` rebias path and the LTS lookup-table path) are fixed for each format, with the source sign preserved.

## fp8e4m3 -> bf16

Bytes `0x7F`/`0xFF` (exponent and mantissa both all-ones) decoded to a finite `±480.0` instead of NaN. Both converters now detect the reserved encoding and select the canonical bf16 quiet NaN (`0x7FC0`).

- **`Fp8E4M3Nv_to_Bf16`** (fmul path): per-lane `icmp` on the masked i16 magnitude, `select` `0x7FC0` for the fmul result. Kept in the per-lane i16 domain to dodge the existing IGC flush-to-zero bug.
- **`Fp8E4M3Nv_to_Bf16Table`**: per-lane `icmp` on the masked i32 value, `select` `0x7FC00000` after the existing subnormal selects.

## fp8e5m2 -> bf16

The Inf encodings (`0x7C`/`0xFC`) decoded to `±65536.0` and the NaN encodings (`0x7D`-`0x7F` / `0xFD`-`0xFF`) to other finite values. Both converters now detect the exponent-all-ones field and substitute the IEEE special bf16 patterns — `±Inf` (`0x7F80`) or quiet NaN (`0x7FC0`).

- **`Fp8E5M2_to_Bf16`** (fmul path): detect on the raw fp8 byte (`byte & 0x7C == 0x7C`) and fix up the extracted scalar lanes downstream of the fmul, leaving the i32-domain FTZ-workaround shape untouched.
- **`Fp8E5M2_to_Bf16Table`**: detect on the masked i32 value (`c & 0x0F800000 == 0x0F800000`), substitute `0x7F800000` / `0x7FC00000` after the existing subnormal selects.

## Behavior change

- fp8e4m3: bytes `0x7F`/`0xFF` now yield NaN (was `±480.0`).
- fp8e5m2: bytes `0x7C`/`0xFC` now yield `±Inf`, and `0x7D`-`0x7F` / `0xFD`-`0xFF` yield NaN (were finite values).

Cost: +2 ops/lane in the fmul paths, +3/element in the table paths. Each format's bit-exact pytest golden and FileCheck lit test are updated.

Fixes #7771

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 1]
```diff
@@ -685,17 +685,40 @@ Fp8E4M3Nv_to_Bf16Table(Location loc, ConversionPatternRewriter &rewriter,
   Value d2 = b.add(i32_ty, c2, b.i32_val(0x3c000000));
   Value d3 = b.add(i32_ty, c3, b.i32_val(0x3c000000));
 
+  // Reserved fp8 bytes 0x7F/0xFF (exponent and mantissa both all-ones) must
+  // decode to the canonical bf16 quiet NaN (0x7FC0), not to the +-480.0
+  // that `d0`..`d3` above would otherwise produce (e.g. for byte 0x7F,
+  // d = c + 0x3c000000 == 0x43F00000, i.e. bf16 0x43F0 == 480.0). `cN` holds
+  // `(byte & 0x7F) << 20`, so the exponent (bits 23-26) and top 3 mantissa
```
[the remark this answers is not in the document]
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:740]: Not accurate — this PR does update Fp8E5M2_to_Bf16Table and Fp8E5M2_to_Bf16 too (hunks at lines 147 and 222), fixing up the reserved Inf/NaN encodings for E5M2 as well, matching the PR description's "fp8e5m2 -> bf16" section.

**Review by chengjunlu — APPROVED** [review]
